healthcareai/trained_models/trained_supervised_model.py

```diff
@@ -183,17 +188,37 @@ def prepare_and_subset(self, dataframe):
             pandas.core.frame.DataFrame: A dataframe that has been run through the pipeline and subsetted to only the columns the model expects.
         """
 
+        # Add response column if not included in prediction data
+        if self.prediction_column not in dataframe.columns.values:
+            dataframe[self.prediction_column] = np.NaN
+
         try:
             # Raise an error here if any of the columns the model expects are not in the prediction dataframe
+            df2 = dataframe.copy()
+            if self.pre_dummified_columns is not None:
+                df2 = df2[self.pre_dummified_columns]
+
+            # Change the dtype of the categorical columns in the prediction dataframe to 'category' with levels determined
+            # by the training data
+            if self.categorical_column_info is not None:
+                for column in self.categorical_column_info:
+                    col_categories = self.categorical_column_info[column].index
+                    df2[column] = df2[column].astype('category', categories=col_categories)
+                    # Check whether the prediction data contains categories not present in the training set and print
+                    # a message warning that these new values will be dropped and imputed
+                    new_values = {v for v in dataframe[column].unique() if not (v in col_categories or pd.isnull(v))}
+                    if len(new_values) > 0:
+                        category_message = 'Column {} contains levels not seen in the training set. These levels have been '
```

Comment:
Awesome! I'll go ahead and clean that up.

Somewhat related: Do you think the message about new levels being removed would be better as a warning (rather than just a print statement)?